api_tests/e2e/rfc003/btc_eth-erc20/test.js

```diff
@@ -86,6 +89,8 @@ describe("RFC003: Bitcoin for ERC20", () => {
                 alpha_ledger_refund_identity: null,
                 beta_ledger_redeem_identity: alice_final_address,
                 alpha_ledger_lock_duration: 144,
+                counterparty_ip_address: bob.host,
```

Comment:
What about `peer` instead of `counterparty`? Peer terminology is commonly used in **peer**-to-**peer** protocols. Counterparty, not so much.

I would not separate the host and port, I find it odd.
You could try to deserialize to `SocketAddr` and if it fails try to deserialize to `IpAddr` with default port.

Reply:
It'd be good to use [ToSocketAddrs](https://doc.rust-lang.org/std/net/trait.ToSocketAddrs.html) actually because it gives you DNS resolution.

I don't think custom deserialization is too bad. Let me dig a bit more see if I can find something better.

Reply:
I think a quick an easy custom derive is fine.